UPGRADE.md

```diff
@@ -1,10 +1,36 @@
 # Upgrade Guide
 
-_After `7.4.0` all `MINOR` and `MAJOR` version bumps will have upgrade notes 
-posted here._
+_`MAJOR` version bumps will have upgrade notes posted here._
+
+[2020-09-28] 7.x.x to 8.x.x
+-----------------------------
+
+### Overview
+
+Version `8.x.x` is the first version that officially drops support for Java 7. 
+
+- Removal of dependencies offering functionality included in Java 8 and beyond:
+    - [Guava concurrency, hashing, and charstreams](https://github.com/twilio/twilio-java/pull/575)
+        - `com.google.common.hash.Hash*` -> `org.apache.commons.codec.digest.DigestUtils`
+        - `com.google.common.io.CharStreams` -> `org.apache.commons.io.IOUtils`
+        - `com.google.common.util.concurrent.*` -> `java.util.concurrent`
+    - [Guava `com.google.common.collect.Range` removal]()
+        - Example: `setMessageDate(final Range<LocalDate> rangeMessageDate)` -> `setMessageDate(final LocalDate messageDateBefore, final LocalDate messageDateAfter)`
+    - [joda-time](https://github.com/twilio/twilio-java/pull/572)
+        - `org.joda.time.DateTime` -> `java.time.ZonedDateTime`
+        - `org.joda.time.LocalDate` -> `java.time.LocalDate`
+- [Removal of deprecated classes and methods](https://github.com/twilio/twilio-java/pull/578):
+    - `com.twilio.jwt.accesstoken.ConversationsGrant`
+    - `com.twilio.jwt.accesstoken.IpMessagingGrant`
+    - `com.twilio.jwt.accesstoken.VideoGrant.getConfigurationProfileSid()`
+    - `com.twilio.jwt.accesstoken.VideoGrant.setConfigurationProfileSid()`
+    - Renamed child TwiML methods:
+        - `com.twilio.twiml.voice.Refer.Builder.referSip`
+        - `com.twilio.twiml.voice.Say.Builder.ssml<Element>`
```

Comment:
What are the old methods/classes that I need to migrate off and what are the new methods/classes that I need to migrate to?

Reply:
Even though all those old methods/classes have been marked as deprecated with a documented workaround for over a year?

Reply:
The docs regarding the workaround are being removed, so adding here or a link will help.